src/main/java/com/aws/greengrass/clientdevices/auth/iot/IotAuthClient.java

```diff
@@ -117,43 +123,51 @@ public Optional<Certificate> getIotCertificate(String certificatePem) throws Inv
         }
 
         @Override
-        @SuppressWarnings("PMD.AvoidCatchingGenericException")
         public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate) {
+            if (Objects.isNull(certificate)) {
```

Comment:
Override this so we can have a version that just needs the certificateId. In cases we don't have the pem but we do have the id (when loaded from the registry)